src/NATS.Client.Core/NatsConnection.cs

```diff
@@ -265,7 +265,7 @@ private async ValueTask InitialConnectAsync()
                     {
                         // upgrade TcpConnection to SslConnection
                         var sslConnection = conn.UpgradeToSslStreamConnection(Opts.TlsOpts, _tlsCerts);
-                        await sslConnection.AuthenticateAsClientAsync(uri).ConfigureAwait(false);
+                        await sslConnection.AuthenticateAsClientAsync(FixTlsHost(uri)).ConfigureAwait(false);
```

Comment:
`FixTlsHost` uses `_lastSeedConnectUri!.Host` which is not set until `SetupReaderWriterAsync` is called on line 298.  Fixing the host up is only needed for reconnects, where IP Addresses have been gossiped but the original hostname of the Seed Server needs to be tracked

Reply:
Good catch! Fixed.